packages/e2e-tests/src/specs/editor/fontCheck/fontCheck.js

```diff
@@ -0,0 +1,156 @@
+  await page.waitForSelector('[role="dialog"]');
+  await expect(page).toMatchElement('button', { text: 'Open anyway' });
+  await expect(page).toMatchElement('button', { text: 'Replace font' });
+  await takeSnapshot(page, 'Missing fonts dialog');
+  await expect(page).toClick(
+    "div[role='dialog'] div[class^='datalist__'] button"
+  );
+  await page.keyboard.type(fontFamily);
+  await page.keyboard.press('ArrowDown');
+  await page.keyboard.press('Enter');
+  await expect(page).toClick('button', { text: 'Replace font' });
+  await expect(page).toClick('[data-testid="textFrame"]');
+  await expect(page).toClick('button[aria-label="Font family"]');
+};
+
+const replaceFontUsingDefault = async () => {
+  await page.waitForSelector('[role="dialog"]');
+  await expect(page).toClick('button', { text: 'Open anyway' });
+  await expect(page).toClick('[data-testid="textFrame"]');
+  await expect(page).toClick('button[aria-label="Font family"]');
+};
+
+const storyWithFontCheckDialog = async (
+  title,
+  customFont = 'Open Sans Condensed Light'
+) => {
+  // take steps needed to get Missing Font dialog to show
+  await addCustomFont();
+  await createStoryWithTitle(title);
+  await updateFont(customFont);
+  await publishStory();
+  await removeCustomFont();
+  await visitStoryWithTitle(title);
+  await page.waitForSelector('[data-testid="textFrame"]');
+  await page.waitForSelector('[role="dialog"]');
+};
+
+describe('Font Check', () => {
+  describe('Font Check Dialog', () => {
+    withExperimentalFeatures(['customFonts']);
+    withExperimentalFeatures(['notifyDeletedFonts']);
+
+    it('should show dialog & replace font with default font', async () => {
+      const title = 'Test replace missing font with (default) Roboto';
+      const replacementFont = 'Roboto';
+      await storyWithFontCheckDialog(title);
+      await replaceFontUsingDefault(replacementFont);
+      const selectedText = await page.$eval(
+        'div[role="dialog"] li[aria-selected="true"]',
+        (el) => el.textContent
+      );
+      await expect(selectedText).toBe(replacementFont);
+    });
+
+    it('should show dialog & replace it with selected font', async () => {
+      const title = 'Test replace missing font with Bungee';
+      const replacementFont = 'Bungee';
+      await storyWithFontCheckDialog(title);
+      await replaceFontWithFontPicker(replacementFont);
+      const selectedText = await page.$eval(
+        'div[role="dialog"] li[aria-selected="true"]',
+        (el) => el.textContent
+      );
+      await expect(selectedText).toBe(replacementFont);
+    });
+
+    it('should show dialog & visit settings page', async () => {
+      const title = 'Test visit to settings from dialog';
+      await storyWithFontCheckDialog(title);
+      await Promise.all([
+        expect(page).toClick('[role="dialog"] a', { text: 'Settings' }),
+        page.waitForNavigation(),
+      ]);
+      await takeSnapshot(page, 'back-to-settings');
```

Comment:
Shouldn't these titles be a little user friendly?

Reply:
Removed those -- forgot I had those in for troubleshooting the e2e tests